rust/src/lib.rs

```diff
@@ -303,6 +334,20 @@ impl WorkQueue {
         db.llen(&self.processing_key)
     }
 
+    pub async fn get_queue_lengths<'a, C: AsyncCommands>(
+        &'a self,
+        db: &mut C,
+    ) -> RedisResult<Vec<i64>> {
```

Comment:
Definitely don't return a `Vec`.

A `Vec` is a heap allocated array + capacity. So a `Vec` itself has 3 ints, the base pointer, the length and the capacity. Further to that, it performs a heap allocation of the data.

If we just pass a tuple (`(i64, i64)`) then its just two ints big. Smaller than the `Vec` pointer itself and no heap allocation!!!

Finally, the lengths aren't signed, so this should be `u64` not `i64`.

Reply:
@Oxygen588 could you please resolve this?

You just need to change the return type to `(u64, u64)` (or `u32`) and then return the result as a tuple. Definitely not a Vec!